### ISSUE TITLE
Issue when changing vm_hostname

### ISSUE THREAD

**Issue by gfediere** [smurawski/chef-zero-scheduled-task#16]
Hi,

When using this provisioner with  vm_hostname option (to change hostname), i don't have any console return :
       Creating the scheduled task.
       SUCCESS: The scheduled task "chef-tk" has successfully been created.
       Transferring files to <pdc-windows-2012r2-standard-chef-12-4-1>
       SUCCESS: Attempted to run the scheduled task "chef-tk".

**Comment by smurawski** [comment]
@gfediere I don't think anything in the provisioner cares about the vm_hostname, but this may be an issue with winrm-transport or the transport layer in test-kitchen.  

Could you upload a run with debug details?

```
kitchen converge -l debug
```

That'd give me a better chance at localizing the problem.

**Comment by gfediere** [comment]
Hi @smurawski ,

Here is the debug:
'D      [FileTransporter] Uploaded 6 items in (0m20.59s)
D      Transfer complete
D      Serializing $env:temp
D      [WinRM] plaintext::http://127.0.0.1:55985/wsman<{:disable_sspi=>true, :basic_auth_only=>true, :user=>"vagrant", :pass=>"vagrant"}> (
            "`$env:temp = '$env:temp'" |
                out-file $env:temp/kitchen/env.ps1
)
D      [WinRM] plaintext::http://127.0.0.1:55985/wsman<{:disable_sspi=>true, :basic_auth_only=>true, :user=>"vagrant", :pass=>"vagrant"}> (
        try {
          $npipeServer = new-object System.IO.Pipes.NamedPipeServerStream(
            'task', [System.IO.Pipes.PipeDirection]::In)
          $pipeReader = new-object System.IO.StreamReader($npipeServer)
          schtasks /run /tn "chef-tk"
          $npipeserver.waitforconnection()
          $host.ui.writeline('Connected to the scheduled task.')
          while ($npipeserver.IsConnected) {
            $output = $pipereader.ReadLine()
            if ($output -like 'SCHEDULED_TASK_DONE:*') {
              $exit_code = ($output -replace 'SCHEDULED_TASK_DONE:').trim()
            }
            else { $host.ui.WriteLine($output) } } }
        finally {
          $pipereader.dispose()
          $npipeserver.dispose()
          $host.setshouldexit($exit_code)
        }
)
       SUCCESS: Attempted to run the scheduled task "chef-tk".
'
Converge stays stuck here

**Comment by smurawski** [comment]
@gfediere I'll need the full debug log from create through converge

**Comment by dpetzel** [comment]
I've also bumped into a similar situation today. In my case I am not changing the hostname. It'll take some work to get a scrubbed debug output, but in the meantime I see this in tail end of the debug output:

```
$ExecutionContext.InvokeCommand.ExpandString($Script) | out-file "$env:temp/kitchen/chef-client-script.ps1"
try {
  Add-Type -AssemblyName System.Core
  $npipeServer = new-object System.IO.Pipes.NamedPipeServerStream('task', 
    [System.IO.Pipes.PipeDirection]::In)
  $pipeReader = new-object System.IO.StreamReader($npipeServer)
  schtasks /run /tn "chef-tk"
  $npipeserver.waitforconnection()
  $host.ui.writeline('Connected to the scheduled task.')
  while ($npipeserver.IsConnected) {
    $output = $pipereader.ReadLine()
    if ($output -like 'SCHEDULED_TASK_DONE:*') {
      $exit_code = ($output -replace 'SCHEDULED_TASK_DONE:').trim()
    }
    else { $host.ui.WriteLine($output) } } }
finally {
  $pipereader.dispose()
  $npipeserver.dispose()
  $host.setshouldexit($exit_code)
}
)
       SUCCESS: Attempted to run the scheduled task "chef-tk".
D      [WinRM] retrying receive request after timeout
D      [WinRM] retrying receive request after timeout

```

In my case this is running Windows 2008R2, TK 1.6.0 and `gem 'chef-zero-scheduled-task', '~> 0.0.12'`

**Comment by dpetzel** [comment]
My issue turned out to be a little different (albeit the original symptom seems the same). I created a different issue for mine at https://github.com/smurawski/chef-zero-scheduled-task/issues/25

**Comment by smurawski** [comment]
I've released v0.2.0 of this gem which deprecates its use and suggests the use of the `elevated` configuration on the transport.
